soccer/gameplay/situational_play_selection.py

```diff
@@ -300,47 +319,67 @@ def isPileup(self):
 
     ##Returns if we are in the specified situation without regard to capitalization
     #If the check variable is true, it will check if the situation exists and throw an
-    #exception is it does now.
-    def isSituation(self, situation, check=False):
-        up = situation.upper()
-
-        if (check):
-            found = False
-            for g in self.Situation:
-                if (up == g):
-                    found = True
-                    break
-            if not found:
-                raise Exception("Passed situation " + situation + " / " + up +
-                                " is not an existing situation")
-
-        if (self.currentSituaion.name == up):
-            return True
+    #exception is it does now
+    # @param situation a situation as a string or a enum 
+    # @param check Strings will be checked if they are real if True
+    def isSituation(self, situation, check=True):
+
+        if (isinstance(situation, str)):
```

Comment:
Is this the expected branch? Shouldn't the situation always be an enum value or am I missing something.